### PR TITLE
Docu invariants tori

### PR DESCRIPTION
**Pull request by wdecker**
[docu invariants tori](https://github.com/oscar-system/Oscar.jl/pull/3428/commits/27e012b18656a928d1fa91f56da9dc225d70ca39)

Preview: https://docs.oscar-system.org/previews/PR3428/InvariantTheory/tori/

### CODE REVIEW

File: docs/doc.main [hunk 1]
```diff
@@ -57,7 +57,6 @@
         "Nemo/qadic.md",
      ],
      "Nemo/finitefield.md",
-     "Nemo/algebraic.md",
```
Comment (lgoettgens) [inline docs/doc.main:60]: This change seems unrelated and may have been introduced by mistake?

File: docs/doc.main [hunk 2]
```diff
@@ -57,7 +57,7 @@
         "Nemo/qadic.md",
      ],
      "Nemo/finitefield.md",
-     "Nemo/algebraic.md", 
+    "Nemo/algebraic.md",
```
Comment (lgoettgens) [inline docs/doc.main]:
```suggestion
     "Nemo/algebraic.md",
```

File: experimental/InvariantTheory/src/TorusInvariantsFast.jl [hunk 3]
```diff
@@ -206,3 +370,42 @@ function torus_invariants_fast(W::Vector{Vector{ZZRingElem}}, R::MPolyRing)
...
+3-element Vector{MPolyDecRingElem}:
+ X[1]^2*X[3]
+ X[1]*X[2]*X[3]
+ X[2]^2*X[3]
+
+julia> invariant_ring_as_affine_algebra_(RT)
+(Quotient of multivariate polynomial ring by ideal (-t[1]*t[3] + t[2]^2), Hom: quotient of multivariate polynomial ring -> graded multivariate polynomial ring)
+```
+"""
+function invariant_ring_as_affine_algebra(R::TorGrpInvRing)
```
Comment (lgoettgens) [inline experimental/InvariantTheory/src/TorusInvariantsFast.jl]: I think in the "usual spirit" of naming functions (and the last bullet point in [Naming conventions](https://docs.oscar-system.org/dev/DeveloperDocumentation/styleguide/#Naming-conventions)), this function should rather be named `affine_algebra(R::TorGrpInvRing)`, as the first part is already clear from the type.

File: experimental/InvariantTheory/src/TorusInvariantsFast.jl [hunk 4]
```diff
@@ -206,3 +370,42 @@ function torus_invariants_fast(W::Vector{Vector{ZZRingElem}}, R::MPolyRing)
         end
     end
 end
+
+#####################Invariant rings as affine algebras
+
+@doc raw"""
+    invariant_ring_as_affine_algebra(RT::TorGrpInvRing)
+
+Return the invariant ring `RT` as an affine algebra.
```
Comment (lgoettgens) [inline experimental/InvariantTheory/src/TorusInvariantsFast.jl]: the second return value is missing in the docstring

**Comment by wdecker** [comment]
This is ready to be merged.

**Comment by lgoettgens** [comment]
> This is ready to be merged.

who can review this?

File: experimental/InvariantTheory/src/TorusInvariantsFast.jl [hunk 5]
```diff
@@ -96,12 +187,84 @@ mutable struct InvariantRingFastTorus
...
+    poly_ring(RT::TorGrpInvRing)
+
+# Examples
+```jldoctest
+julia> T = torus_group(QQ,2)
+Torus of rank 2
+  over QQ
+```
+"""
+poly_ring(R::TorGrpInvRing) = R.poly_ring
```
Comment (joschmitt) [inline experimental/InvariantTheory/src/TorusInvariantsFast.jl:221]: For the invariants of finite groups, we call this `polynomial_ring` by now. (This is not really a flaw with this pull request.)

**Review by joschmitt — APPROVED** [review]
Looks good to me.
(Unfortunately, I don't have the rights to give an actual approving review.)

**Comment by thofma** [comment]
Maybe @joschmitt could have a quick glance?

**Review by fieker — APPROVED** [review]

**Comment by thofma** [comment]
Yeah, I don't think we want a `poly_ring` function floating around.

**Comment by joschmitt** [comment]
> Yeah, I don't think we want a `poly_ring` function floating around.

See #3442. Note that there is still
```
poly_ring(R::AbstractAlgebra.Generic.AbsMSeriesRing)
```

**Comment by thofma** [comment]
Yes, this should be unexported.
